### PR TITLE
Translate JVM crash report page into French

### PR DESCRIPTION
**Pull request by Xav83**
Hi Jenkins devs :slightly_smiling_face:

I am creating this Pull Request to help, during the Hacktoberfest, with the french translation.

Adds the french translation of the property files in the "core/src/main/resources/jenkins/diagnosis/HsErrPidList" folder.

### Testing done

Using GitPod, I have accessed the page `HsErrPidList` by making it always be activated locally, then, I runned Jenkins, downloaded the `Locale` plugin, changed the language to `fr` in the System configuration menu, and finally, I clicked on the message telling me to look at the logfiles.
Screenshot from 2022-11-09 09-36-13

 

### Proposed changelog entries

* Translate JVM crash report page into French.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7253"></a>

### CODE REVIEW

**Review by basil — COMMENTED** [review]
Thank you for your submission! This is highly appreciated.

The testing section is mandatory, not optional. Please test this PR as described in the PR template and write a **Testing done** section.

When you test this change, you will discover that it has a (minor) bug. Identifying and fixing the bug is left as an exercise for the reader to be performed when testing the change.

**Comment by Xav83** [comment]
Hi @basil :-)

I have added missing single quote as it was probably the source of the minor bug you mentioned.
I am currently trying to run the tests with Gitpod, but I am experimenting a lot of failing tests which don't seems to come from the file I have added.
Do you have an idea about which test I could run in order to be sure about the modification of this Pull Request ?

**Review by NotMyFault — APPROVED** [review]
Read fine so far, with my little understatement of French.

I requested a native speaker for review, maybe we can make use of your talent, @gounthar 🙏🏻

**Comment by gounthar** [comment]
I can't get it to work with gitpod because of `[ERROR] Medium: Switch statement found in hudson.util.QuotedStringTokenizer.hasMoreTokens() where default case is missing [hudson.util.QuotedStringTokenizer] At QuotedStringTokenizer.java:[lines 156-271] SF_SWITCH_NO_DEFAULT` but I'll make some proposals in my review.

File: core/src/main/resources/jenkins/diagnosis/HsErrPidList/description_fr.properties
Comment (gounthar) [inline core/src/main/resources/jenkins/diagnosis/HsErrPidList/description_fr.properties:1]:
```suggestion
blurb = Informe de la présence de fichiers de vidage d''incidents JRE après une fermeture anormale de Jenkins, et permet de les voir.
```

File: core/src/main/resources/jenkins/diagnosis/HsErrPidList/index_fr.properties
Comment (gounthar) [inline core/src/main/resources/jenkins/diagnosis/HsErrPidList/index_fr.properties:1]:
```suggestion
blurb=Les rapports d''incident JVM suivants ont été trouvés pour cette instance de Jenkins. \
```

File: core/src/main/resources/jenkins/diagnosis/HsErrPidList/index_fr.properties
Comment (gounthar) [inline core/src/main/resources/jenkins/diagnosis/HsErrPidList/index_fr.properties:2]:
```suggestion
  Si vous pensez que c''est un problème interne à Jenkins, merci de <a href="https://www.jenkins.io/redirect/report-an-issue">le reporter</a>. \
```

File: core/src/main/resources/jenkins/diagnosis/HsErrPidList/index_fr.properties
Comment (gounthar) [inline core/src/main/resources/jenkins/diagnosis/HsErrPidList/index_fr.properties:3]:
```suggestion
  Jenkins s''appuie sur des heuristiques afin de trouver ces fichiers. Pour une découverte plus fiable envisagez s''il vous plaît d''ajouter \
```

File: core/src/main/resources/jenkins/diagnosis/HsErrPidList/index_fr.properties
Comment (gounthar) [inline core/src/main/resources/jenkins/diagnosis/HsErrPidList/index_fr.properties:4]:
```suggestion
  <code>-XX:ErrorFile=/path/to/hs_err_pid%p.log</code> en tant qu''argument de la JVM.
```

File: core/src/main/resources/jenkins/diagnosis/HsErrPidList/message_fr.properties
Comment (gounthar) [inline core/src/main/resources/jenkins/diagnosis/HsErrPidList/message_fr.properties:1]:
```suggestion
blurb=Cette instance Jenkins semble avoir rencontré un incident logiciel. <a href="{0}">Merci de vérifier les fichiers de logs</a>.
```

**Review by gounthar — CHANGES_REQUESTED** [review]
Thanks a bunch for this translation.
See my proposals in the review.
Please let me know if you have any questions about my feedback!

**Comment by NotMyFault** [comment]
Hey @Xav83,

do you mind taking a look at the feedback, please?

**Comment by Xav83** [comment]
Hi @gounthar @NotMyFault,

I have integrated the elements from the @gounthar's review :slightly_smiling_face:

**Review by gounthar — APPROVED** [review]
Looks good to me, thanks for your work.

**Review by NotMyFault — APPROVED** [review]
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Review by basil — COMMENTED** [review]
Why have the `blurb` and `ago` keys been translated in `core/src/main/resources/jenkins/diagnosis/HsErrPidList/index_fr.properties` but not the `Java VM Crash Reports` (i.e., the title), `Name`, `Date`, and `Delete` keys? How was this change tested? Please provide a screenshot of this page rendered in French to prove that you have tested this change.

**Comment by basil** [comment]
> I am not sure to understand your latest comment.

You are to understand my latest comment as a statement that your submission is incomplete for two reasons:

1. It only translates half of the localizable messages for the `HsErrPidList` page, rather than all of them.
2. The mandatory **Testing Done** section of your pull request is blank.

**Comment by Xav83** [comment]
Hi @basil,

I am currently experimenting with Gitpod and Jenkins and I have finally successfully open Jenkins interface in Gitpod \o/

I am now facing the problem to access the `HsErrPidList` page, in order to be able to see the issue you are referencing (and to validate the fix, once I will have implemented it). Do you have any advice to help me with this point ?

Thanks in advance for your time and your help on this Pull Request :pray:

**Comment by basil** [comment]
https://github.com/jenkinsci/jenkins/blob/79c60432981b5b62159748b463e1b0b33fc5fc97/core/src/main/java/jenkins/diagnosis/HsErrPidList.java#L166-L169 triggers the administrative monitor based on the presence of a Java crash error file, so if I could not induce a crash I would temporarily make that method return true just to get the administrative monitor to show up, then remove that change once I was done testing.

**Comment by Xav83** [comment]
Thank you @basil for all those information! I have been able to access the `HsErrPidList` page, so I made the modification about the keys `Date`, `Delete`, `Name` and `Java VM Crash Reports`.
Screenshot from 2022-11-09 09-36-13

File: core/src/main/resources/jenkins/diagnosis/HsErrPidList/index_fr.properties
Comment (basil) [inline core/src/main/resources/jenkins/diagnosis/HsErrPidList/index_fr.properties:9]: What does JM stand for? The French translation of Java VM (i.e., Virtual Machine) is [Machine virtuelle Java](https://fr.wikipedia.org/wiki/Machine_virtuelle_Java), so I cannot see what the "J" in "JM" would stand for (since Java already appears before the acronym in your translation).

Comment (Xav83) [reply core/src/main/resources/jenkins/diagnosis/HsErrPidList/index_fr.properties:9]: Oops! This is a typo ! --'
I meant to type "VM"...

Indeed as you suggested I can tranlate the acronyme 'VM' into french ! I will do it tonight 🙂

Comment (basil) [reply core/src/main/resources/jenkins/diagnosis/HsErrPidList/index_fr.properties:9]: I do not think we need to introduce an inconsistency by having an acronym in the English version and a full word in the French version. But I do not think we should introduce a typo either.

**Review by basil — APPROVED** [review]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
